lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,434 @@
+				.ok_or(LightningError {
+					err: "Couldn't find channel for update".to_owned(),
+					action: ErrorAction::IgnoreError,
+				})?;
+
+			let directional_info = channel
+				.get_directional_info(channel_flags)
+				.ok_or(LightningError {
+					err: "Couldn't find previous directional data for update".to_owned(),
+					action: ErrorAction::IgnoreError,
+				})?;
+
+			let htlc_maximum_msat =
+				if let Some(htlc_maximum_msat) = directional_info.htlc_maximum_msat {
+					OptionalField::Present(htlc_maximum_msat)
+				} else {
+					OptionalField::Absent
+				};
+
+			UnsignedChannelUpdate {
+				chain_hash,
+				short_channel_id,
+				timestamp: latest_seen_timestamp,
+				flags: standard_channel_flags,
+				cltv_expiry_delta: directional_info.cltv_expiry_delta,
+				htlc_minimum_msat: directional_info.htlc_minimum_msat,
+				htlc_maximum_msat,
+				fee_base_msat: directional_info.fees.base_msat,
+				fee_proportional_millionths: directional_info.fees.proportional_millionths,
+				excess_data: vec![],
+			}
+		};
+
+		if channel_flags & 0b_0100_0000 > 0 {
+			let cltv_expiry_delta: u16 = Readable::read(read_cursor)?;
+			synthetic_update.cltv_expiry_delta = cltv_expiry_delta;
+		}
+
+		if channel_flags & 0b_0010_0000 > 0 {
+			let htlc_minimum_msat: u64 = Readable::read(read_cursor)?;
+			synthetic_update.htlc_minimum_msat = htlc_minimum_msat;
+		}
+
+		if channel_flags & 0b_0001_0000 > 0 {
+			let fee_base_msat: u32 = Readable::read(read_cursor)?;
+			synthetic_update.fee_base_msat = fee_base_msat;
+		}
+
+		if channel_flags & 0b_0000_1000 > 0 {
+			let fee_proportional_millionths: u32 = Readable::read(read_cursor)?;
+			synthetic_update.fee_proportional_millionths = fee_proportional_millionths;
+		}
+
+		if channel_flags & 0b_0000_0100 > 0 {
+			let tentative_htlc_maximum_msat: u64 = Readable::read(read_cursor)?;
+			synthetic_update.htlc_maximum_msat = if tentative_htlc_maximum_msat == u64::max_value()
+			{
+				OptionalField::Absent
+			} else {
+				OptionalField::Present(tentative_htlc_maximum_msat)
+			};
+		}
+
+		network_graph.update_channel_unsigned(&synthetic_update)?;
+	}
+
+	let input_termination_check: Result<u8, DecodeError> = Readable::read(read_cursor);
+	if let Err(DecodeError::ShortRead) = input_termination_check {
+		Ok(())
+	} else {
+		Err(GraphSyncError::ProcessingError(
+			"unexpected data found".to_string(),
+		))
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use std::thread;
+	use std::time::Duration;
+
+	use bitcoin::blockdata::constants::genesis_block;
+	use bitcoin::Network;
+
+	use lightning::ln::msgs::DecodeError;
+	use lightning::routing::network_graph::NetworkGraph;
+
+	use crate::error::GraphSyncError;
+	use crate::processing::update_network_graph;
+
+	#[test]
+	fn network_graph_fails_to_update_from_clipped_input() {
+		let block_hash = genesis_block(Network::Bitcoin).block_hash();
+		let network_graph = NetworkGraph::new(block_hash);
+
+		// OldestDataDirection0, recent timestamp, with last byte removed
```

Comment:
What does `OldestDataDirection0` refer to? Same for similar comments in other tests.

Reply:
This is a comment I left for myself that I use in combination with some experimental variables in the server code to generate these reference updates whenever the format changes. I'd prefer to take out these comments in a follow-up PR.